File: labs/lab5_6/Lab2_GeometricFigures/Square.cs
using System;

namespace Lab2_GeometricFigures
{
    public class Square : Rectangle, IPrint
    {
        public Square(double side) : base(side, side)
        {
        }

        public double Side
        {
            get { return Width; }
            set
            {
                if (value <= 0)
                    throw new ArgumentException("Сторона квадрата должна быть положительным числом");

                Width = value;
                Height = value;
            }
        }

        public override string ToString()
        {
            return $"Квадрат: Сторона = {Side:F2}, Площадь = {Area():F2}";
        }
    }
}

File: labs/lab2/lab_python_oop/Square.py
from lab_python_oop.Rectangle import Rectangle

class Square(Rectangle):
    name = "Квадрат"

    def __init__(self, side, color):
        super().__init__(side, side, color)

    def __repr__(self):
        return "Фигура: {0}, сторона: {1}, цвет: {2}, площадь: {3}".format(
            self.getName(), self.width, self.colorObj, self.area())
